tasks/run-typedoc-on-updates.ts

```diff
@@ -0,0 +1,66 @@
+import { execSync } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import { isArray } from 'lodash';
+import { sync as glob } from 'glob';
+
+function readFile(filename: string): any {
+	const content = fs.readFileSync(filename, 'utf8');
+	return JSON.parse(content);
+}
+
+function executeGitCommand(command: string) {
+	return execSync(command)
+		.toString('utf8')
+		.replace(/[\n\r\s]+$/, '')
+		.split('\n');
+}
+
+const resolveFilePaths = (paths?: unknown) => {
+	if (isArray(paths)) {
+		return (paths ?? []).map((fileDir: string) =>
+			path.resolve(__dirname, '..', fileDir),
+		);
+	}
+	console.error('Paths should be array');
+	return [];
+};
+
+const committedFilePaths = executeGitCommand('git diff --name-only main');
+
+const committedRelativeFilePaths = resolveFilePaths(committedFilePaths);
+
+const TYPEDOC_DIR = path.resolve(__dirname, '../typedoc.json');
+
+const typeDocInput: string[] = readFile(TYPEDOC_DIR)?.entryPoints ?? [];
+
+const getDocumentedFiles = () => {
```

Comment:
I think you can simplify to:
```ts
const documentedFiles = typeDocInput.flatMap(
  entryPoint => glob(entryPoint, { realpath: true },
));
```

Reply:
What does `documentedFiles` mean? Files that contain the documentation or files that are documented?

Reply:
entryPoint files for `typedoc` (that matches the pattern)

Reply:
> entryPoint files for typedoc (that matches the pattern)

Sorry, what I meant: could you adjust the variable to communicate clearer the purpose of it.